.github/workflows/build-viz-state.yml

```
@@ -0,0 +1,19 @@
+on:
+  workflow_call:
+
+jobs:
+  viz-state:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/download-artifact@v3
+        with:
+          name: latest-result
+      - run: rustup toolchain install stable --profile minimal
```

Here we can use caching and our pre-defined environment (includes a specific rust revision).
```suggestion
      - uses: Swatinem/rust-cache@v2
      - uses: cachix/install-nix-action@v19
        with:
          github_access_token: ${{ secrets.GITHUB_TOKEN }}
```